File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/impl/metamodel/hashmap/aas/security/Security.java
package org.eclipse.basyx.aas.impl.metamodel.hashmap.aas.security;

import java.util.HashMap;

import org.eclipse.basyx.aas.api.metamodel.aas.security.ISecurity;
import org.eclipse.basyx.aas.impl.metamodel.facades.SecurityFacade;

/**
 * KeyElements as defined in DAAS document
 * 
 * @author schnicke
 *
 */
public class Security extends HashMap<String, Object> implements ISecurity {
	private static final long serialVersionUID = 1L;

	public static final String ACCESSCONTROLPOLICYPOINTS = "accessControlPolicyPoints";
	public static final String TRUSTANCHOR = "trustAnchor";

	/**
	 * Constructor
	 */
	public Security() {
		// Default values
		put(ACCESSCONTROLPOLICYPOINTS, null);
		put(TRUSTANCHOR, null);
	}

	@Override
	public Object getAccessControlPolicyPoints() {
		return new SecurityFacade(this).getAccessControlPolicyPoints();
	}

	public void setAccessControlPolicyPoints(Object obj) {
		new SecurityFacade(this).setAccessControlPolicyPoints(obj);
	}

	@Override
	public Object getTrustAnchor() {
		return new SecurityFacade(this).getTrustAnchor();
	}

	public void setTrustAnchor(Object obj) {
		new SecurityFacade(this).setTrustAnchor(obj);
	}
}

File: sdks/c++/basys.sdk.cc/src/aas/aas/map/security/Security.cpp
/*
 * Security.cpp
 *
 *      Author: wendel
 */

#include "BaSyx/aas/map/security/Security.h"

namespace basyx {
namespace aas {

Security::Security(ISecurity & other)
  : vab::ElementMap()
{}

Security::Security(basyx::object obj)
  : vab::ElementMap(obj)
{}

basyx::object Security::getAccessControlPolicyPoints() const
{
  return this->map.getProperty(ISecurity::Path::AccessControlPolicyPoints);
}

basyx::object Security::getTrustAnchor() const
{
  return this->map.getProperty(ISecurity::Path::TrustAnchor);
}

void Security::setAccessControlPolicyPoints(const basyx::object & obj)
{
  this->map.insertKey(ISecurity::Path::AccessControlPolicyPoints, obj);
}

void Security::setTrustAnchor(const basyx::object & obj)
{
  this->map.insertKey(ISecurity::Path::TrustAnchor, obj);
}

}
}